File: FR_14_04_Karty/KARTY.rb
def wynik(x)
    p={ '2'=>2,
        '3'=>3,
        '4'=>4,
        '5'=>5,
        '6'=>6,
        '7'=>7,
        '8'=>8,
        '9'=>9,
        'T'=>10,
        'J'=>11,
        'Q'=>12,
        'K'=>13,
        'A'=>14}
    return p[x]
end
j=gets()
s=gets()
wj = 0
ws = 0
j.strip().split('').each do |x|
    wj=wj+wynik(x)
end
s.strip().split('').each do |x|
    ws=ws+wynik(x)
end

if wj>ws 
    print "JASIO"
elsif wj<ws
    print "STASIO"
else
    print "REMIS"
end

File: FR_14_04_Karty/KARTY.f90
program karty
    implicit none
    interface
        integer function wynik(x)
            character :: x
            character(len=13) :: wszystkie
        end function wynik
    end interface
    
    
    character(len=52) :: j, s
    integer :: wj, ws
    integer i
    wj=0
    ws=0
    read(*,*) j
    read(*,*) s
    do i=1,len_trim(j)
        wj =wj+ wynik(j(i:i))
    end do
    do i=1,len_trim(s)
        ws =ws+ wynik(s(i:i))
    end do

    if (wj>ws) then 
        write(*,*) 'JASIO'
    else if (wj<ws) then
        write(*,*) 'STASIO'
    else 
        write(*,*) 'REMIS'
    end if 
    
    write(*,*) wj
    end program karty
    integer function wynik(x)
        character :: x
        character(len=13) :: wszystkie
        
        wszystkie = "23456789TJQKA"
        wynik= index(wszystkie, x)+1
    end function wynik